### PR TITLE
add current_epoch to dumped_params

### LINKED ISSUE #3260: auto_scale_batch_size won't reset current_epoch
## 🐛 Bug

When `auto_scale_batch_size` is enabled, the model is initially trained with varying batch sizes. When training begins, `trainer.current_epoch` equals 1 instead of 0.

### To Reproduce

Either observe the progress bar or use a simple callback to track the epoch number, once with  `auto_scale_batch_size`  enabled and once with  `auto_scale_batch_size` disabled.

```
from pytorch_lightning import Callback

class PrintCallback(Callback):
    
    def __init__(self):
        self.observed_epochs = []
        
    def on_train_epoch_start(self, trainer, pl_module):
        print(f'Current Epoch: {trainer.current_epoch}')
        self.observed_epochs.append(trainer.current_epoch)

```

### PR DESCRIPTION
**Pull request by maxjeblick**
## What does this PR do?

Fixes #3260 by restoring the current_epoch after finishing the batch size finder.

### CODE REVIEW

**Comment by rohitgr7** [comment]
`global_step`, `total_batch_idx`, or maybe more needs to be reset too. I believe a better solution would be to make a `pre_fit` method or something in `Trainer` and reassign them to their default values every time `.fit` is called by calling it in `.fit` itself.

**Comment by awaelchli** [comment]
@rohitgr7 There is a Tuner class coming, #3160 in which I believe the resetting problem will be tackled.

**Comment by rohitgr7** [comment]
@awaelchli will it support user-defined callbacks?

**Comment by awaelchli** [comment]
I'm not sure, but we recently added support to persist state of callbacks, so we should be able to dump their state too and restore them easily if that's the problem? For further questions better ask directly here #3160

**Comment by maxjeblick** [comment]
The Tuner class looks really promising!

**Comment by SkafteNicki** [comment]
Just to update, the `HyperTuner` class is not going to happen at the moment, instead the tuner algorithms will be moved to a separate `trainer.tune()` method instead of being called by `fit` (it has already started https://github.com/PyTorchLightning/pytorch-lightning/pull/3293). So we need to fix this problem when the refactor is done.

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@maxjeblick mind add a test to check that it is fixed...

**Review by SkafteNicki — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -101,6 +101,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed counter-intuitive error being thrown in `Accuracy` metric for zero target tensor ([#3764](https://github.com/PyTorchLightning/pytorch-lightning/pull/3764))
 
+- Fixed Truner dump: add `current_epoch` to dumped_params ([#3261](https://github.com/PyTorchLightning/pytorch-lightning/pull/3261))
```
Comment (awaelchli) [inline CHANGELOG.md:104]:
```suggestion
- Fixed Tuner dump: add `current_epoch` to dumped_params ([#3261](https://github.com/PyTorchLightning/pytorch-lightning/pull/3261))
```

**Review by awaelchli — APPROVED** [review]
there are probably lots of other trainer attributes that need to be dumped, for example global_step?

**Review by justusschock — APPROVED** [review]

**Comment by SkafteNicki** [comment]
@awaelchli you are probably right. I think the tuner algorithms needs an refactoring after v1.0.

**Comment by rohitgr7** [comment]
Shouldn't tuner be refactored in such a way that it won't call .fit again? Then we won't need to dump and restore anything. What it should do is just suggest and user should change the values and rerun?

**Comment by SkafteNicki** [comment]
@rohitgr7 we use `fit` internally in these algorithms so we don´t have to redefine the training loop necessary to run these algorithms. However, IMO moving forward we should refactor it in such a way that tuner should just create a new `trainer` that can be used during tuning and then destroyed afterwards.

**Comment by rohitgr7** [comment]
@SkafteNicki I am not suggesting to rewrite anything. The reason we need to dump and restore is because these attributes are not re-initialized and current workflow is something like:
```python
trainer.tuner.lr_find()
# suggest
trainer.fit()
```
What I am suggesting is:
```python
trainer.tuner.lr_find()
# suggest
# and tell user to reinitalize trainer, LM, and LDM(if any) with the updated hparams from the suggestions.
```

 > IMO moving forward we should refactor it in such a way that tuner should just create a new trainer that can be used during tuning and then destroyed afterwards.

this is good alternative but this will work for only Trainer and not for LM and LDM since there are chances that some attributes might be changed during `.fit()`.

**Pull request merged**